docs/integration.md

```diff
@@ -0,0 +1,466 @@
+
+## Database
+
+[Liquidbase](https://www.liquibase.org/) is used to manage Chutney RDBMS schema.  
+You can find corresponding changelog [here](https://github.com/chutney-testing/chutney/blob/master/server/src/main/resources/changelog/db.changelog-master.xml).
+
+!!! note
+    Chutney has been tested with H2 and PostgreSQL databases.
+
+To configure your datasource, use the property `spring.datasource`
+
+=== "H2 (memory)"
+    ``` yaml
+    spring:
+        datasource:
+            url: jdbc:h2:mem:dbName
+    ```
+    !!! note
+        You can find an example in maven module [local-dev](https://github.com/chutney-testing/chutney/tree/master/packaging/local-dev), which uses an embedded H2 with filesystem persistence.
+
+=== "PostgreSQL (SSL two way)"
+    ``` yaml
+    spring:
+        datasource:
+            url: jdbc:postgresql://host:port/dbName?ssl=true&sslmode=verify-ca&sslfactory=org.postgresql.ssl.DefaultJavaSSLFactory&currentSchema=mySchema
+            username: user
+    ```
+
+## Authentication & Permissions
+
+!!! important
+    Maven module [local-dev](https://github.com/chutney-testing/chutney/tree/master/packaging/local-dev) shows :
+    
+    * How to use in memory authentication and roles, see the `mem-auth` profile.
+    * How to use a custom LDAP authentication (for example purpose, it uses an embedded LDAP server).
+
+Chutney uses Spring Security for :
+
+* Basic authentication
+* Enforce authentication and check authorization on API (ex. admin rights Spring Boot [Actuator](#spring-boot-actuator) endpoints)
+* Configuring in memory users and roles with a Spring profile [mem-auth](link) if needed
+
+
+??? note "How to use in memory Spring profile `mem-auth`"
+    * Activate the profile
+
+    ``` yaml
+    spring:
+        profiles:
+            active:
+              - mem-auth
+    ```
+
+    * Declare users and roles
+
+    ``` yaml
+    chutney:
+      security:
+        users:
+          -
+            id: user
+            name: user
+            firstname: user
+            lastname: user
+            mail: user@user.com
+            password: user-password-bcrypt
+            roles: # (1)!
+              - role-with-admin-in-it
+    ```
+    
+    1. Optional, if the role include the characters 'admin', ignoring case, all permissions will be granted to that user.
+
+!!! warning
+    If you create a role name including characters 'admin' (ignoring case), all permissions will be granted to users with this role.
+
+If you want to add another authentication mechanism, you should follow the [Spring security architecture](https://spring.io/guides/topicals/spring-security-architecture).
+
+!!! important "Authentication requirements"
+    The principal build by the authentication mechanism must be an instance of the Chutney [UserDto](https://github.com/chutney-testing/chutney/blob/master/server/src/main/java/com/chutneytesting/security/api/UserDto.java).
```

Comment:
I don't understand what you mean

Reply:
Spring Security relies on an interface (UserDetails i think) which represents a user. This one must be implemented to specify what is a user for a specific application. For Chutey, it's the UserDto class.
If an authentiation mechanism create a Userdetails instance of another type, some Chutney functionalities break.